docs/architecture/dapr-for-net-developers/serviceinvocation.md

```diff
@@ -0,0 +1,347 @@
+
+``` csharp
+public class CatalogService : ICatalogService
+{
+    private const string DaprAppId = "catalog-api";
+
+    private readonly DaprClient _daprClient;
+
+    public CatalogService(DaprClient daprClient)
+    {
+        _daprClient = daprClient;
+    }
+
+    public async Task<IEnumerable<CatalogItem>> GetCatalogItemsAsync(
+        IEnumerable<int> ids)
+    {
+        return await _daprClient.InvokeMethodAsync<IEnumerable<CatalogItem>>(
+            DaprAppId,
+            "api/v1/catalog/items",
+            new HttpInvocationOptions
+            { 
+                Method = HttpMethod.Get,
+                QueryString = new Dictionary<string, string>
+                {
+                    ["ids"] = string.Join(",", ids)
+                }
+            });
+    }
+
+    // ...
+}
+```
+
+The `HttpInvocationOptions` object is used to pass the query string and HTTP verb details to the Catalog API.
+
+The other call made by the aggregator service is to the Basket API. It only allows authorized requests. The access token is passed along in an *Authorization* request header to ensure the call succeeds:
+
+``` csharp
+public class BasketService : IBasketService
+{
+    private const string DaprAppId = "basket-api";
+
+    public Task UpdateAsync(BasketData currentBasket, string accessToken)
+    {
+        return _daprClient.InvokeMethodAsync(
+            DaprAppId,
+            "api/v1/basket",
+            currentBasket,
+            new HttpInvocationOptions
+            {
+                Headers = new Dictionary<string, string>
+                {
+                    [HeaderNames.Authorization] = accessToken
+                }
+            });
+    }
+
+    // ...
+}
+```
+
+There is no need to explicitly specify the HTTP method for this POST call as `Post` is the default value of `HttpInvocationOptions.Method`.
+
+## Summary
+
+In this chapter, you learned about the service invocation building block. You saw how to invoke remote methods both by making direct HTTP calls to the Dapr sidecar, and by using the Dapr .NET SDK.
+
+The eShopOnDapr reference architecture shows how the original eShopOnContainers solution is modernized by using Dapr service invocation. Adding Dapr to eShop provides benefits such as automatic retries, message encryption using mTLS, and improved observability.
+
+### References
+
+- [Dapr service invocation building block](https://docs.dapr.io/developing-applications/building-blocks/service-invocation/)
+
+- [Monitoring distributed cloud-native applications](https://docs.microsoft.com/en-us/dotnet/architecture/cloud-native/observability-patterns)
+
+>[!div class=“step-by-step”]
+>[Previous](~index.md~)
+>[Next](~index.md~)
```

Comment:
There are build warnings, where the file ~index.md~ cannot be resolved. Let's fix these two please.

Reply:
I've commented out the page navigation section for now. We'll push a separate PR to make filenames consistent and fix navigation across all chapters.